### PR TITLE
SCRUM-72: refactor FetchPruducts

### PR DESCRIPTION
**Pull request by SanaO200**

### CODE REVIEW

File: src/services/fetchCarousel.js [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-		if (!response.ok) {
-			throw new Error(`HTTP error! Status: ${response.status}`);
-		}
-		const data = await response.json();
-		setCarouselItems(data.carouselItems);
-	} catch (err) {
-		console.log('Error: ', err.message);
-	}
+  try {
+    const { data } = await axios.get(`/api/v1/${carouselCategory}`)
```
Comment (filkovsp) [inline src/services/fetchCarousel.js:3]: missing import for axios, 
this makes application crashed when trying to render carousel component

File: src/services/fetchCarousel.js [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+// 		const response = await fetch(`/api/v1/${carouselCategory}`);
+// 		if (!response.ok) {
+// 			throw new Error(`HTTP error! Status: ${response.status}`);
+// 		}
+// 		const data = await response.json();
+// 		setCarouselItems(data.carouselItems);
+// 	} catch (err) {
+// 		console.log('Error: ', err.message);
+// 	}
+// }
```
Comment (filkovsp) [inline src/services/fetchCarousel.js:23]: not needed, if migrated to axios

File: src/containers/NewProducts/NewProducts.jsx [hunk 3]
```diff
@@ -1,43 +1,57 @@
...
+          : 'Завантаження...'}
+      </div>
+      <div className="d-flex justify-content-end">
+        <button onClick={showMoreProducts} className={styles.buttonShowMore}>
+          Показати ще
+        </button>
+      </div>
+    </div>
+  )
 }
```
Comment (filkovsp) [inline src/containers/NewProducts/NewProducts.jsx:57]: try this instead:
```jsx
import { useState, useEffect } from 'react'
import ApiClient from '../../services/apiClient'
import ProductCard from '../../components/ProductCard/ProductCard'
import styles from './NewProducts.module.scss'

export default function NewProducts() {
  const [products, setProducts] = useState([])
  const [offset, setOffset] = useState(0)
  const [loading, setLoading] = useState(false)
  const filter = 'is_new'
  const limit = 4

  const fetchMoreProducts = async () => {
    const params = { filter, offset, limit }
    const uri = '/v1/products'
    try {
      setLoading(true)
      const { data } = await ApiClient.get(uri, { params })
      setProducts(previousProducts => [...previousProducts, ...data.products])
    } catch (error) {
      console.error('Error: ', error.message)
    } finally {
      setLoading(false)
    }
  }

  const increaseOffset = () => {
    setOffset(previousOffset => previousOffset + limit)
  }

  useEffect(() => {
    if (!loading) {
      fetchMoreProducts()
    }
  }, [offset])

  return (
    <div className='container'>
      <h2>Новинки </h2>
      <div className='row'>
        {products?.length > 0 &&
          products.map(newProduct => {
            return (
              <div className='col-lg-3 col-md-4 col-sm-6 col-12 mb-4' key={newProduct.id}>
                <div className='rounded flex-column h-100'>
                  <ProductCard itemCard={newProduct} />
                </div>
              </div>
            )
          })}
        {loading && 'Завантаження...'}
      </div>
      <div className='d-flex justify-content-end'>
        <button onClick={increaseOffset} className={styles.buttonShowMore}>
          Показати ще
        </button>
      </div>
    </div>
  )
}
```
and plus, return `StrictMode` back in `main.jsx`

File: src/main.jsx [hunk 4]
```diff
@@ -3,8 +3,4 @@ import ReactDOM from 'react-dom/client'
 import App from './containers/App'
 import './styles/main.scss'
 
-ReactDOM.createRoot(document.getElementById("root")).render(
-  <StrictMode>
-    <App />
-  </StrictMode>
-);
+ReactDOM.createRoot(document.getElementById('root')).render(<App />)
```
Comment (filkovsp) [inline src/main.jsx:6]:
```suggestion
ReactDOM.createRoot(document.getElementById("root")).render(
  <StrictMode>
    <App />
  </StrictMode>
);
```

plus, check comment for src/containers/NewProducts/NewProducts.jsx

File: src/services/fetchCarousel.js (hunk 1 above)
Comment (SanaO200) [reply src/services/fetchCarousel.js:3]: ok

File: src/services/fetchCarousel.js (hunk 2 above)
Comment (SanaO200) [reply src/services/fetchCarousel.js:23]: ok

File: src/containers/NewProducts/NewProducts.jsx (hunk 3 above)
Comment (SanaO200) [reply src/containers/NewProducts/NewProducts.jsx:57]: new proposal- to use: useEffect(() => {
       if (!loading & (productsRef.current === false)) {
      fetchProducts()
        return () => {
        productsRef.current = true
      }
    }
  }, [])

**Review by filkovsp — APPROVED** [review]

**Pull request merged**